### ISSUE TITLE
Undefined index.

### ISSUE THREAD

**Issue by theantichris** [lifo101/typeahead-bundle#13]
Hey, running into a bit of a snag.

I have the controller setup to return the JSON object, that bit is working.

I have the following code in my form object:

```
->add('localAccountExecs', 'entity_typeahead', [
    'label'  => 'Local Account Executive(s)',
    'class'  => 'AssociateBundle:Associate',
    'render' => 'name',
    'route'  => 'associate_list',
            ])
```

My template looks like:

```
{{ form_row(form.localAccountExecs, {attr: {placeholder: 'Search for an associate...'}}) }}
```

When loading the form I get the following error:

Notice: Undefined index: 0000000063caf225000000000ef8ef71

**Comment by lifo101** [comment]
Are you returning a 'name' field in your JSON response?


-- Jason

**Comment by theantichris** [comment]
I am. Here's a sample of the response.
```
[
  {
    "id": 1,
    "name": "Barry"
  },
  {
    "id": 2,
    "name": "Kristy"
  },
  {
    "id": 3,
    "name": "Sam"
  },
  {
    "id": 4,
    "name": "Amber"
  }
]
```

**Comment by lifo101** [comment]
I forgot to mention that you still need to return a 'value' field with the
value you want to render (the same as 'name' in your case). The 'render'
option is used elsewhere. It's a little confusing and should be refactored
at some point.


-- Jason

**Comment by theantichris** [comment]
My JSON object now looks like:

```
{
    "id": 1,
    "name": "Barry",
    "value": "Barry"
  }
```

Still getting the undefined index error.

**Comment by lifo101** [comment]
Well, that's strange. I don't know what the undefined index of 0000000063caf225000000000ef8ef71 would be w/o debugging it in the browser. Can you debug it in your browser using Firebug, or whatever you have in your browser?

**Comment by theantichris** [comment]
At the risk of sounding dump I've never debugged PHP using something like Firebug, just JavaScript.

Here's a stack track if that helps.

```
in vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php at line 2879   + 

at ErrorHandler ->handleError ('8', 'Undefined index: 000000001d051ca3000000003fcb38da', '/home/vagrant/Projects/Averitt Express/Lotus-Notes-Migration/vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php', '2879', array('entity' => object(ArrayCollection))) 
in vendor/doctrine/orm/lib/Doctrine/ORM/UnitOfWork.php at line 2879   + 

at UnitOfWork ->getEntityIdentifier (object(ArrayCollection)) 
in vendor/lifo/typeahead-bundle/Lifo/TypeaheadBundle/Form/DataTransformer/EntityToPropertyTransformer.php at line 40   + 

at EntityToPropertyTransformer ->transform (object(ArrayCollection)) 
in vendor/symfony/symfony/src/Symfony/Component/Form/Form.php at line 1160   + 

at Form ->normToView (object(ArrayCollection)) 
in vendor/symfony/symfony/src/Symfony/Component/Form/Form.php at line 353   + 

at Form ->setData (object(ArrayCollection)) 
in vendor/symfony/symfony/src/Symfony/Component/Form/Extension/Core/DataMapper/PropertyPathMapper.php at line 57   + 

at PropertyPathMapper ->mapDataToForms (object(CorporateAccount), object(RecursiveIteratorIterator)) 
in vendor/symfony/symfony/src/Symfony/Component/Form/Form.php at line 396   + 

at Form ->setData (object(CorporateAccount)) 
in vendor/symfony/symfony/src/Symfony/Component/Form/Form.php at line 488   + 

at Form ->initialize () 
in vendor/symfony/symfony/src/Symfony/Component/Form/FormBuilder.php at line 226   + 

at FormBuilder ->getForm () 
in vendor/symfony/symfony/src/Symfony/Component/Form/FormFactory.php at line 39   + 

at FormFactory ->create (object(CorporateAccountType), object(CorporateAccount)) 
in src/AppBundle/CorporateAccountBundle/Controller/CorporateAccountController.php at line 136   + 

at CorporateAccountController ->handleAccountForm (object(CorporateAccount)) 
in src/AppBundle/CorporateAccountBundle/Controller/CorporateAccountController.php at line 104   +
 
at CorporateAccountController ->createAction () 
at call_user_func_array (array(object(CorporateAccountController), 'createAction'), array()) 
in app/bootstrap.php.cache at line 3109   + 

at HttpKernel ->handleRaw (object(Request), '1') 
in app/bootstrap.php.cache at line 3071   + 

at HttpKernel ->handle (object(Request), '1', true) 
in app/bootstrap.php.cache at line 3222   + 

at ContainerAwareHttpKernel ->handle (object(Request), '1', true) 
in app/bootstrap.php.cache at line 2444   + 

at Kernel ->handle (object(Request)) 
in web/index.php at line 14   + 
```

**Comment by theantichris** [comment]
Think I got somewhere. I was going over the documentation again and added the multiple option to the form. Getting a new error now that sounds more  hopeful.

```Key "input_group" for array with keys "placeholder, data-prototype" does not exist in LifoTypeaheadBundle:Form:typeahead.html.twig at line 46
500 Internal Server Error - Twig_Error_Runtime```

**Comment by theantichris** [comment]
This only seems to be an issue when trying to use the multiple option for the form field. Your bundle works perfectly with multiple set to false.

Is there an implementation step I'm missing?

**Comment by theantichris** [comment]
I just overrode the Twig template and removed

```
{% if attr.input_group is not empty %}style="display:none"{% endif %}
```

Everything seems to work perfectly.

**Comment by lifo101** [comment]
Are you using the MopaBootstrapBundle? That attribute comes from that
bundle.


-- Jason

**Comment by theantichris** [comment]
We're not. We implemented our own Bootstrap themes.

**Comment by lifo101** [comment]
then that must be the issue. I didn't realize I was referencing an
attribute from the other bundle that isn't technically required for my
bundle to function.


-- Jason

**Comment by lifo101** [comment]
Fixed in [v2.3.2](https://github.com/lifo101/typeahead-bundle/releases/tag/2.3.2)
